### PR TITLE
Kazaz/add models

### PR DESCRIPTION
**Pull request by kazazor**
# What this PR is all about?

* Added models to handle the data we'll need for: https://coinmarketcap.com/api/
* Removed duplicate settings for our URLs in webpack
* Removed the `getLanguage` function from `i18n`. Everything should be fetched from the store
* Changed all the warnings in sass lint to be errors
* Changed the theme of the site to be dark (I think people like it better when it comes to things like that.. Makes them feel more professional IMO. In any case we could modify it later)

Now it looks like this:
screen shot 2017-06-16 at 9 18 10 pm

I took the idea from a trading website I'm using which also uses react material-ui:
screen shot 2017-06-16 at 9 17 17 pm

### Additional information for the reviewer

There was a "Deaf Zone" created at Grateful Dead concerts where deaf patrons held onto inflated balloons to feel the vibration of the music.

# Question

@OmriAharon -  Do you think we need UT for the models..? Or is it too simple at the moment / we want to run faster?

## Making sure I did my homework

Please check all the notes you took care of. You should have them all checked.
**As a reviewer**, if you see something not checked, please do not `P-LGTM` or `LGTM` the PR.

- [X] **Cross browser** - I checked that my code works on: FF, Safari (>=10.0) & Chrome.
- [X] **Mobile responsiveness** - I checked that my code works on the above browser also in mobile view.
- [] **Unit Tests** - I covered and updated the unit tests according to this commit changes (if necessary).

### CODE REVIEW

File: src/components/Site.js [hunk 1]
```diff
@@ -30,7 +30,7 @@ class Site extends Component {
   }
 
   render() {
-    const isRTL = getLanguage().isRTL;
+    const isRTL = this.props.locale && this.props.locale.isRTL;
```
Comment (OmriAharon) [inline src/components/Site.js]: `this.props.locale &&` is going to be a recurring code segment I feel, we can add an empty object for it in the reducer and then anywhere we can just query `this.props.locale.isRTL`.

To do that, we can just replace the initial state line in siteReducer:

`const initialState = Immutable.from({ locale: {} });`

File: src/components/Site.js [hunk 2]
```diff
@@ -39,8 +39,16 @@ class Site extends Component {
   }
 }
 
+const mapStateToProps = (state, ownProps) => ({
```
Comment (OmriAharon) [inline src/components/Site.js]: NIT -> redundant `ownProps`

File: src/models/currencies/BTC.js [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import Currency from './Currency';
+
+export default class BTC extends Currency {
+  constructor() {
+    super('BTC', 'Bitcoin', 'BTC', Currency.SYMBOL_LOCATIONS.END);
```
Comment (OmriAharon) [inline src/models/currencies/BTC.js]: Eh? How does this work? I don't see you have a constructor that receives 4 parameters, only one object.. What am I missing? 😕

File: src/styles/base/_colors.scss [hunk 4]
```diff
@@ -0,0 +1,5 @@
+// We should use as few as web can here. All themes should be defined in the material-ui js theme.
```
Comment (OmriAharon) [inline src/styles/base/_colors.scss]: web -> we

File: src/styles/base/_normalize.scss [hunk 5]
```diff
@@ -1,3 +1,10 @@
+// We should use as few as web can here. All themes should be defined in the material-ui js theme.
```
Comment (OmriAharon) [inline src/styles/base/_normalize.scss]: web -> we

File: src/helpers/getSiteTheme.js [hunk 6]
```diff
@@ -8,7 +8,7 @@ import {
   cyan700, grey600, pink700, pink500, pink400, fullWhite
 } from 'material-ui/styles/colors';
 
-const primaryColor = '#262626';
+const primaryColor = '#383838';
```
Comment (kazazor) [inline src/helpers/getSiteTheme.js]: The colors is only when we have too.. In general everything should be used from the theme here, only needed it for the background color

File: src/models/currencies/BTC.js (hunk 3 above)
Comment (kazazor) [inline src/models/currencies/BTC.js]: Oops.. never push when you have to leave the house :)

File: src/components/Site.js (hunk 1 above)
Comment (kazazor) [inline src/components/Site.js]: Good idea

**Pull request merged**